app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt

```diff
@@ -664,6 +665,40 @@ class PasswordStore : AppCompatActivity(R.layout.activity_pwdstore) {
         }.launch(intent)
     }
 
+    private fun renameCategoryAux(oldCategory: PasswordItem, error: Boolean) {
+        val view = layoutInflater.inflate(R.layout.folder_dialog_fragment, null)
+        val newCategoryEditText = view.findViewById<TextInputEditText>(R.id.folder_name_text)
+        if (error) {
+            newCategoryEditText.error = getString(R.string.message_error_rename_folder)
+        }
+
+        MaterialAlertDialogBuilder(this)
+            .setTitle(R.string.title_rename_folder)
+            .setView(view)
+            .setMessage(getString(R.string.message_rename_folder, oldCategory.name))
+            .setPositiveButton(R.string.dialog_ok) { _, _ ->
+                val newCategory = File("${oldCategory.file.parent}/${newCategoryEditText.text}")
+                when {
+                    newCategoryEditText.text.isNullOrBlank() ||
+                        newCategory.exists() -> renameCategoryAux(oldCategory, true)
+                    else -> {
+                        newCategory.mkdirs()
+                        oldCategory.file.renameTo(newCategory)
+                        AutofillMatcher.updateMatches(this, mapOf(oldCategory.file to newCategory))
+                        commitChange(resources.getString(R.string.git_commit_move_text, oldCategory.name, newCategory.name))
+                    }
+                }
+            }
+            .setNegativeButton(R.string.dialog_cancel, null)
+            .show()
```

Comment:
```suggestion
            .create()
        dialog.requestInputFocusOnView<TextInputEditText>(R.id.folder_name_text)
        dialog.show()
```

This should make the keyboard show up automatically when the dialog comes up. My formatting might be off so just fix it up in Android Studio.